lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -261,6 +266,61 @@ def connect_ws(opts={}, timeout = 20)
     socket.extend(Rex::Proto::Http::WebSocket::Interface)
   end
 
+  # Defines a proc to log HTTP requests
+  #
+  # @param http_trace [Bool] A Boolean representing datastore['HttpTrace'] to check if HttpTrace is set or unset.
+  # @param http_trace_headers_only [Bool] A Boolean representing datastore['HttpTraceHeadersOnly'] to check if only HTTP headers need to be logged.
+  # @param http_trace_colors [String] A string representing datastore['HttpTraceColors'] to specify the colors in which request and response need to be logged.
+  # @return [Proc] A Proc object to log HTTP requests, or nil if datastore['HttpTrace'] is unset
+  def set_http_trace_proc_request(http_trace, http_trace_headers_only, http_trace_colors)
```

Comment:
It looks like the implementation of this is quite similar to the login scanner; Could we extract these into a single implementation? That way folk can change the implementation in one place

```ruby
class HttpSubscriber
  def on_request(request)
  end

  def on_response(response)
  end
end

class HttpLoggerSubscriber < HttpSubscriber
  def initialize(logger: logger)
    @logger = logger
  end

  def on_request(request)
    logger.print_line(".... request ...")
  end

  def on_response(response)
    logger.print_line(".... response ...")
  end
end
```

Usage:

```ruby
http_logger_subscriber = HttpLoggerSubscriber.new(logger: self)
```

Note that in the scenario of a login scanner - we will pass in `logger: framework_module` here - whilst in the context of running a module we'll pass in `logger: self` which will allow for the implementation to be consistent across both usages.

We'll also update the http client instantiation as well:

```ruby
    nclient = Rex::Proto::Http::Client.new(
      ... params as usual...,
      subscriber: http_logger_subscriber,
   )
```